src/ot_orchestration/dags/platform_input_support.py

```diff
@@ -1,96 +1,102 @@
-PIS_MACHINE_SPEC = {"cpu": "1", "memory": "512Mi"}
-
-
-def get_steps() -> list[str]:
-    """Read the steps from the PIS configuration file."""
-    yaml_config = read_yaml_config(PIS_CONFIG_PATH)
-    return yaml_config["steps"].keys()
 
+PIS_IMAGE = "europe-west1-docker.pkg.dev/open-targets-eu-dev/platform-input-support-test/platform-input-support-test"
+PIS_SACC = "platform-input-support@open-targets-eu-dev.iam.gserviceaccount.com"
+CONFIG_PATH = Path(__file__).parent / "config" / "pis.yaml"
+CONFIG = read_yaml_config(CONFIG_PATH)
+GCS_URL = CONFIG["gcs_url"]
 
 with DAG(
-    "platform_input_support",
+    dag_id="platform_input_support",
     default_args=shared_dag_args,
     description="Open Targets Platform — platform-input-support",
     **platform_dag_kwargs,
+    params={
+        "run": Param(
+            f"pis-{datetime.now().strftime('%Y%m%d-%H%M')}",
+            description="Unique run identifier, will be added as a label to infrastructure resources.",
+        ),
+        "pis_image_tag": Param(
+            "latest",
+            description=(
+                "The tag of the platform-input-support image to use. Defaults to 'latest', "
+                "but it is recommended to use a specific tag. Keep in mind tags do not have the 'v' prefix."
+            ),
+        ),
+    },
 ) as dag:
-    for step_name in get_steps():
-        clean_step_name = clean_name(step_name)
+    for step in list(CONFIG["steps"].keys()):
+
+        @task_group(group_id=f"pis_{step}")
+        def pis_step(step: str) -> None:
+            instance_name = f"uo-pis-{clean_name(step)}-{{{{ run_id | strhash }}}}"
+            config_filename = f"config-{clean_name(step)}.yaml"
+
+            c = PISDiffComputeOperator(
+                task_id=f"diff_{step}",
+                step_name=step,
+                local_config_path=CONFIG_PATH,
+                upstream_config_url=f"{GCS_URL}/{config_filename}",
+            )
+
+            u = UploadFileOperator(
+                task_id=f"upload_config_{step}",
+                src=CONFIG_PATH,
+                dst=GCS_URL + "/" + config_filename,
+            )
 
-        with TaskGroup(group_id=clean_step_name):
+            r = ComputeEngineRunContainerizedWorkloadSensor(
+                task_id=f"run_{step}",
+                instance_name=instance_name,
+                labels={"subteam": "backend", "tool": "pis", "run": dag.params["run"]},
+                container_image=f"{PIS_IMAGE}:{dag.params['pis_image_tag']}",
+                container_env={
+                    "PIS_STEP": step,
+                    "PIS_CONFIG_FILE": "/" + config_filename,
+                    "PIS_POOL": "16",
+                },
+                container_service_account=PIS_SACC,
+                container_scopes=["https://www.googleapis.com/auth/drive"],
+                container_files={f"{GCS_URL}/{config_filename}": "/" + config_filename},
+                work_disk_size_gb=150,
```

Comment:
@javfg I wonder if then it could be referenced in the step config.

Reply:
true, that it won't change frequently, but I imagine that every step do not have the same requirement of 150Gb disk usage, am I correct in this assumption?

Reply:
You're correct, but step machines (and their disks) live for average 2 to 3 minutes, to a maximum of 17 minutes for one of them. The cost difference for that time span is zero, as far as I can tell.

Although I cannot say this 100% sure. There is no trace of the disk cost for PIS in the billing even if I'm labeling the disks. This could be because they are added into the VM cost because they are attached disks.

If I'm labeling everything properly, running PIS is about £3.40.